File: app/client/src/types/stock-price.ts
export interface StockPrice {
  symbol: string
  security_name: string
  date: string
  open: number
  high: number
  low: number
  close: number
  adj_close: number
  volume: number
}

export interface StockPricePageResponse {
  data: StockPrice[]
  total: number
  skip: number
  limit: number
}

export interface SymbolInfo {
  symbol: string
  security_name: string
  count: number
}

export interface SymbolListResponse {
  symbols: SymbolInfo[]
}

File: app/backend/models/stock_price.py
"""
MongoDB document shape for the 'stock_prices' collection.

{
    "symbol": str,
    "security_name": str,
    "date": datetime,
    "open": float,
    "high": float,
    "low": float,
    "close": float,
    "adj_close": float,
    "volume": int,
    "import_id": ObjectId
}
"""

STOCK_PRICES_COLLECTION = "stock_prices"

EXPECTED_CSV_COLUMNS = [
    "Symbol",
    "Security Name",
    "Date",
    "Open",
    "High",
    "Low",
    "Close",
    "Adj Close",
    "Volume",
]
